src/ecs/commands/updateEnableExecuteCommandFlag.ts

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as vscode from 'vscode'
+import { extensionSettingsPrefix } from '../../shared/constants'
+import { DefaultSettingsConfiguration } from '../../shared/settingsConfiguration'
+import { recordEcsDisableExecuteCommand, recordEcsEnableExecuteCommand } from '../../shared/telemetry/telemetry.gen'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { EcsServiceNode } from '../explorer/ecsServiceNode'
+
+export async function updateEnableExecuteCommandFlag(
+    node: EcsServiceNode,
+    enable: boolean,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<void> {
+    const hasExecEnabled = node.service.enableExecuteCommand
+
+    const enableWarning = localize(
+        'AWS.command.ecs.runCommandInContainer.warning.enableExecuteFlag',
+        'Enabling command execution will change the state of resources in your AWS account, including but not limited to stopping and restarting the service.\nAltering the state of resources while the Execute Command is enabled can lead to unpredictable results.\n Continue?'
+    )
+    const disableWarning = localize(
+        'AWS.command.ecs.runCommandInContainer.warning.disableExecuteFlag',
+        'Disabling command execution will change the state of resources in your AWS account, including but not limited to stopping and restarting the service.\n Continue?'
+    )
+    const yes = localize('AWS.generic.response.yes', 'Yes')
+    const yesDontAskAgain = localize('AWS.message.prompt.yesDontAskAgain', "Yes, and don't ask again")
+    const no = localize('AWS.generic.response.no', 'No')
+
+    const configuration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+
+    if (enable) {
```

Comment:
Control flow feels a bit goofy: `enable` cases are nested and have to `return` to leave the function, while `disable` isn't nested, still has some early returns, and can run out the function. Can you condense some of this logic? e.g. emit metrics, make API calls, and update the tree in only one location at the end. This would make it easier to add success/fail/cancel results for telemetry.